src/handlers/meta_transaction_handlers.ts

```diff
@@ -232,11 +236,35 @@ export class MetaTransactionHandlers {
                     });
                     return;
                 }
+                // TODO(oskar) - refactor?
+                if (this._rateLimiter !== undefined) {
+                    const { isAllowed, reason } = await this._rateLimiter.isAllowedAsync(apiKey);
+                    if (!isAllowed) {
+                        const ethereumTxn = await this._metaTransactionService.generatePartialExecuteTransactionEthereumTransactionAsync(
+                            zeroExTransaction,
+                            signature,
+                            protocolFee,
+                        );
+                        res.status(HttpStatus.TOO_MANY_REQUESTS).send({
```

Comment:
Todo: we should document that the caller can submit this themselves when rate limited. Janky experience but works better with rfqt then another fresh quote? Or maybe just quote in normal flow. Not sure